### PR TITLE
Driver adapted to build with kernel `6.1.1`

### PR DESCRIPTION
**Pull request by cyring**
This PR is fixing the `remove` function prototype changed in Linux `6.1.1`
It also replaces `module_i3c_i2c_driver` macro with standard module init, exit.

### CODE REVIEW

File: apml_sbtsi.c [hunk 1]
```diff
@@ -367,26 +368,40 @@ static int sbtsi_i2c_probe(struct i2c_client *client,
 	return create_misc_tsi_device(tsi_dev, dev, id);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 1)
+static void sbtsi_i3c_remove(struct i3c_device *i3cdev)
+#else
 static int sbtsi_i3c_remove(struct i3c_device *i3cdev)
+#endif
```
Comment (akky16) [inline apml_sbtsi.c:375]: dd9267034c0e364b49261c3e0070b863286d1242 patch introduced,  remove callback return void for i3c & introduced in version 5.12

File: sbrmi.c [hunk 2]
```diff
@@ -357,26 +358,40 @@ static int sbrmi_i3c_probe(struct i3c_device *i3cdev)
...
 	return 0;
+#endif
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 1)
+static void sbrmi_i3c_remove(struct i3c_device *i3cdev)
+#else
 static int sbrmi_i3c_remove(struct i3c_device *i3cdev)
+#endif
 {
```
Comment (akky16) [inline sbrmi.c:384]: dd9267034c0e364b49261c3e0070b863286d1242 patch introduced, remove callback return void for i3c & introduced in version 5.12

File: apml_sbtsi.c [hunk 3]
```diff
@@ -429,7 +444,19 @@ static struct i2c_driver sbtsi_driver = {
 	.id_table = sbtsi_id,
 };
 
-module_i3c_i2c_driver(sbtsi_i3c_driver, &sbtsi_driver)
```
Comment (akky16) [inline apml_sbtsi.c:432]: This macro is fixed in linux kernel with below commit,
commit 01744ce9f07f0b76b0b2d30adba2a7c104f1ff2a
Author: Naveen Krishna Chatradhi <nchatrad@amd.com>
Date:   Mon Dec 5 10:54:13 2022 +0000

    i3c: Correct the macro module_i3c_i2c_driver

    Present definition for module_i3c_i2c_driver uses only the
    1st argument i.e., struct i3c_driver.
    Irrespective of CONFIG_I3C being enabled/disabled,
    struct i2c_driver is never passed to module_driver()

    Passing struct i2c_driver as the 4th argument works.

    Signed-off-by: Akshay Gupta <Akshay.Gupta@amd.com>
    Signed-off-by: Naveen Krishna Chatradhi <nchatrad@amd.com>
    Link: https://lore.kernel.org/r/20221205105413.937704-1-naveenkrishna.chatradhi@amd.com
    Signed-off-by: Alexandre Belloni <alexandre.belloni@bootlin.com>

File: sbrmi.c [hunk 4]
```diff
@@ -419,7 +434,19 @@ static struct i3c_driver sbrmi_i3c_driver = {
 	.id_table = sbrmi_i3c_id,
 };
 
-module_i3c_i2c_driver(sbrmi_i3c_driver, &sbrmi_driver)
```
Comment (akky16) [inline sbrmi.c:422]: This macro is fixed in linux kernel with below commit,
commit 01744ce9f07f0b76b0b2d30adba2a7c104f1ff2a
Author: Naveen Krishna Chatradhi [nchatrad@amd.com](mailto:nchatrad@amd.com)
Date: Mon Dec 5 10:54:13 2022 +0000

i3c: Correct the macro module_i3c_i2c_driver

Present definition for module_i3c_i2c_driver uses only the
1st argument i.e., struct i3c_driver.
Irrespective of CONFIG_I3C being enabled/disabled,
struct i2c_driver is never passed to module_driver()

Passing struct i2c_driver as the 4th argument works.

Signed-off-by: Akshay Gupta <Akshay.Gupta@amd.com>
Signed-off-by: Naveen Krishna Chatradhi <nchatrad@amd.com>
Link: https://lore.kernel.org/r/20221205105413.937704-1-naveenkrishna.chatradhi@amd.com
Signed-off-by: Alexandre Belloni <alexandre.belloni@bootlin.com>

**Comment by akky16** [comment]
Thank you for the pull request, added comments

**Comment by nchatrad** [comment]
we have submitted a proper fix for the kernel vrsio related issues.

**Pull request closed without merging**